cocotb/regression.py

```diff
@@ -316,36 +378,25 @@ def _result_was():
             else:
                 self.log.error("Test error has lead to simulator shutting us "
                                "down", exc_info=exc_info)
-                self._add_failure(result)
-                self._store_test_result(test.module, test.funcname, False, sim_time_ns, real_time, ratio_time)
-                self.tear_down()
-                return
+                result_pass = False
+            # whether we expected it or not, the simulation has failed unrecoverably
+            sim_failed = True
 
         elif test.expect_error:
             if isinstance(result, test.expect_error):
                 self.log.info("Test errored as expected: " + _result_was())
             else:
-                self.log.info("Test errored with unexpected type: " + _result_was())
-                self._add_failure(result)
+                self.log.error("Test errored with unexpected type: " + _result_was(), exc_info=exc_info)
                 result_pass = False
 
         else:
             self.log.error("Test Failed: " + _result_was(), exc_info=exc_info)
-            if _pdb_on_exception:
-                if sys.version_info >= (3, 5):
-                    traceback = exc_info.__traceback__
-                else:
-                    traceback = exc_info[2]
```

Comment:
Please don't remove compatibility code until we've gotten around to updating `setup.py` to not allow python 2.7

Reply:
Done in #1337

Reply:
Calling `pdb.post_mortem` without args works on both Python 2 and 3, we discussed this in the original PR but it was not addressed. But I guess it doesn't matter now.

Reply:
I don't think that's true - there's no exception in flight here, we're not in an `except` block.

Reply:
fixed